SparkPost/2web2ui
# FE-1085 - Update Hibana Matchbox to 4.0.0-hibana.24 and Matchbox Icons to 1.3.0

[FE-1085](https://jira.int.messagesystems.com/browse/FE-1085)

### What Changed
- Installs the Hibana version of Matchbox, version 23
- Updates the Matchbox icons package to `1.3.0`
- Installs `styled-components` - now a peer dependency of Matchbox
- Updated a few Cypress tests that were failing as a result of this upgrade - Modals more frequently require the `force: true` to click on elements within them
- Removes background on the `SortLabel` component - a bug introduced by a new version of Chrome
- Removed custom positioning within the `ShareModal` component that was fixed with the Matchbox upgrade
 - Used new `disableResponsiveBehavior` prop on the templates tabs to prevent overflowing from rendering a popover
- Removed some hacky CSS that Matchbox fixed with the upgrade

### How To Test
- Verify unit/integration tests continue to pass
- Take a look at release notes to check for potentially breaking changes - they are linked via the Jira ticket
- Click around for anything funky!

### To Do
